### PR TITLE
Add generated client for RH SSO dynamic client API

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Generate the dynamic client API via OpenAPI generate.

The spec was retrieved from `https://sso.redhat.com/auth/realms/redhat-external/apis/openapi.yaml`.
All non-dynamic client API parts of the API were stripped down.

Ultimately, the best way would be to have this be a part of https://github.com/redhat-developer/app-services-sdk-go/tree/main/serviceaccountmgmt and referenced from there.

For the time being, we can keep it here and react on API changes this way.

Next steps would be:
- Create a wrapper that uses this client as well as sets a custom `http.Client` with a custom `http.RoundTripper` that will add the correct authorization header, e.g. via `oauth2.TokenSource`.
- Remove the manually written client under `pgk/client/redhatsso` and replace with the wrapper.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* main
  * **PR #437** <a href="https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/437" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/437?utm_source=stack-comment).

**Comment by misberner** [comment]
The PR is hard to review because of a noisy diff, caused by not inserting adequately formatted generation comments.

It looks like a more recent version of OpenAPI generator has [fixed the header template](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go/partial_header.mustache#L18). I don't know if it would be easy to upgrade to a later version. If not, can you please add the `// Code generated by ...; DO NOT EDIT.` comment in `openapi/templates/model.mustache` instead, right before or after `{{>partial_header}}`?

**Comment by dhaus67** [comment]
@misberner Hm, but this is already there (at the very least for the Go files): https://github.com/stackrox/acs-fleet-manager/pull/437/files#diff-93035861bf2250a3ad5e016e08d0a784cdc73cd0e2cb4df60de14853bd960942R8

Do you want it to be more prominent (like `DO NOT EDIT: Code generatey by XXX` at the top of the header?

**Review by misberner — APPROVED** [review]
Looks great, let's 🚢  it once the build passes

File: pkg/client/redhatsso/api/.openapi-generator/VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+4.3.1
```
[the remark this answers is not in the document]
Comment (dhaus67) [reply pkg/client/redhatsso/api/.openapi-generator/VERSION:1]: It's written by the OpenAPI generator.

**Pull request merged**